lib/appRelationsDiscovery.js

```diff
@@ -48,20 +48,24 @@ async function getRelations(asyncApiDocs, { syntax } = defaultOptions) {
               application = relation.get(channelName);
             } else {
               application = {
-                sub: [],
-                pub: [],
+                sub: new Map(),
+                pub: new Map(),
               };
               relation.set(channelName, application);
             }
     
             const channel = doc.channel(channelName);
             const title = doc.info().title();
-    
+            
             if (channel.hasPublish()) {
-              application.pub.push(title);
+              if (application.pub.has(title))
+                throw new Error(`${title} is already publishing to ${channel}`);
+              application.pub.set(title,channel);
```

Comment:
one-line if here looks terrible 😅 It's hard to see that `application.pub.set(title,channel);` is outside the if. Please, write the body of if in the `{}`

```suggestion
              if (application.pub.has(title)) {
                throw new Error(`${title} is already publishing to ${channel}`);
              }
              application.pub.set(title,channel);
```